homeassistant/components/blink/__init__.py

```diff
@@ -154,26 +155,21 @@ def _async_import_options_from_data_if_missing(
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Blink entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-
-    if not unload_ok:
-        return False
-
-    hass.data[DOMAIN].pop(entry.entry_id)
-
-    if len(hass.data[DOMAIN]) != 0:
-        return True
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+        if not hass.data[DOMAIN]:
+            return True
 
-    hass.services.async_remove(DOMAIN, SERVICE_REFRESH)
-    hass.services.async_remove(DOMAIN, SERVICE_SAVE_VIDEO)
-    hass.services.async_remove(DOMAIN, SERVICE_SEND_PIN)
+        hass.services.async_remove(DOMAIN, SERVICE_REFRESH)
```

Comment:
We don't want to remove services when unloading a config entry nowadays. The services should stay registered and handle missing or not loaded config entry in the service handler. Integration wide services should be registered in `async_setup` before the config entry is setup.

See `fully_kiosk` integration for an example.

Reply:
This one will take me a bit - working on something I broke first.